l2l/utils/JUBE_runner.py

```diff
@@ -25,7 +25,7 @@ def __init__(self, trajectory):
         self.trajectory = trajectory
         self.done = False
         if 'JUBE_params' not in self.trajectory.par.keys():
-            print("exception")
+            raise KeyError("The trajectory must contain the parameter group JUBE_params")
         else:
             args = self.trajectory.par["JUBE_params"].params
```

Comment:
You don't need this if else condition if `KeyError` is being raised anyways, since that would be the default behaviour of accessing a key in `self.trajectory.par` that doesn't exist.

Also, would it be possible to use `self.trajectory.parameters` instead of `self.trajectory.par` everywhere?